datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -479,8 +479,15 @@ fn create_batch_from_map(
                     results[i].push(acc.evaluate(&state_accessor).unwrap());
                 }
             }
-            for scalars in results {
-                columns.push(ScalarValue::iter_to_array(scalars)?);
+            for (scalars, field) in results
+                .into_iter()
+                .zip(output_schema.fields()[columns.len()..].iter())
```

Comment:
nitpick: you might want to give `output_schema.fields()[columns.len()..]` a descriptive name via a `let` binding (and maybe a comment as to why this will not panic); took a while to understand why this is (probably) correct.